src/name.rs

```diff
@@ -379,13 +389,19 @@ fn presented_ip_address_matches_constraint(
 }
 
 #[derive(Clone, Copy)]
-enum NameIteration {
+pub(crate) enum NameIteration {
     KeepGoing,
     Stop(Result<(), Error>),
 }
 
-fn iterate_names(
-    subject: untrusted::Input, subject_alt_name: Option<untrusted::Input>,
+/// Nowadays, the subject is ignored and only SANs are considered when
+/// validating a certificate's "subject".
+///
+/// - https://groups.google.com/a/chromium.org/d/msg/security-dev/IGT2fLJrAeo/csf_1Rh1AwAJ
+/// - https://bugs.chromium.org/p/chromium/issues/detail?id=308330
+/// - https://bugzilla.mozilla.org/show_bug.cgi?id=1245280
+pub(crate) fn iterate_names(
+    subject: Option<untrusted::Input>, subject_alt_name: Option<untrusted::Input>,
```

Comment:
This seems to be misunderstanding how webpki, before this PR, is using `subject`. webpki uses `subject` ONLY to verify that the subject conforms to any directoryName constraints that may be present in the chain. It doesn't do any DNS name or IP address matching of the common name. And, as was mentioned in the initial review of this PR, we don't want to do any such processing of the common name. Instead we only want to look at iPAddress subjectAltNames.

Thus, unless I'm massively understanding what you're trying to accomplish in this PR, everything related to the subject and especially the subject's common name should be removed.